### ISSUE TITLE
Compatability rules for ESLint

### ISSUE THREAD

**Issue by mgeisler** [eslint/eslint#1694]
Hi all,

Due to #1455, I now see this if I try updating to ESLint 0.12.0:
```
% grunt eslint
Running "eslint:target" (eslint) task
Warning: Definition for rule 'space-unary-word-ops' was not found. Use --force to continue.

Aborted due to warnings.
```
(I run it via Grunt, but that's not the problem.)

In #1455, the `space-unary-word-ops` rule was renamed to `space-unary-ops`. I think the rule was made more flexible too, which is probably a good and nice idea. However, changing the rule name is not very friendly for people who want to upgrade to the new version.

Is ESLint still considered unstable as the `0.x.y` versions numbers suggest? Could old rules be made to still work by letting them call the new rule for compatibility? Could the [0.10.0 blog post][1] at least mention this as a breaking change?

[1]: http://eslint.org/blog/2014/11/eslint-0.10.0-released/

**Comment by nzakas** [comment]
Brought up in #1538 and spawned #1549. 

Yes, ESLint is still considered unstable, which is why it's not 1.0.0 yet. The change in 0.10.0 was flagged incorrectly and so didn't make it into the release notes correctly; I'll fix that.

After 0.10.0 we decided to make a concerted effort to not rename or remove rules again until 1.0.0. In the meantime, we will deprecate old rules and add new ones, and the deprecated ones will be removed prior to the 1.0.0 release.

**Comment by mgeisler** [comment]
Thank you, that's all I hoped to hear! Sorry about the duplicate issue.

**Comment by mtsweir** [comment]
Not sure if I'm in the right place to post my issue, when I run grunt I get this error, I have installed the pushy via bower into the project which when I add the jquery to the main.js file I get this error.

This is the bower package file
https://github.com/christophery/pushy 

Error below:

app/scripts/main.js
   4:0   error  "$" is not defined                             no-undef
   4:18  error  Missing "use strict" statement                 strict
   6:1   error  "$" is not defined                             no-undef
   7:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
   7:3   error  Multiple spaces found before 'dots'            no-multi-spaces
   8:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
   8:3   error  Multiple spaces found before 'infinite'        no-multi-spaces
   9:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
   9:3   error  Multiple spaces found before 'speed'           no-multi-spaces
  10:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  10:3   error  Multiple spaces found before 'slidesToShow'    no-multi-spaces
  11:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  11:3   error  Multiple spaces found before 'slidesToScroll'  no-multi-spaces
  12:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  12:3   error  Multiple spaces found before 'responsive'      no-multi-spaces
  13:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  13:5   error  Multiple spaces found before '{'               no-multi-spaces
  14:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  14:7   error  Multiple spaces found before 'breakpoint'      no-multi-spaces
  15:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  15:7   error  Multiple spaces found before 'settings'        no-multi-spaces
  16:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  16:9   error  Multiple spaces found before 'slidesToShow'    no-multi-spaces
  17:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  17:9   error  Multiple spaces found before 'slidesToScroll'  no-multi-spaces
  18:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  18:9   error  Multiple spaces found before 'infinite'        no-multi-spaces
  19:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  19:9   error  Multiple spaces found before 'dots'            no-multi-spaces
  20:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  20:7   error  Multiple spaces found before '}'               no-multi-spaces
  21:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  21:5   error  Multiple spaces found before '}'               no-multi-spaces
  22:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  22:5   error  Multiple spaces found before '{'               no-multi-spaces
  23:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  23:7   error  Multiple spaces found before 'breakpoint'      no-multi-spaces
  24:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  24:7   error  Multiple spaces found before 'settings'        no-multi-spaces
  25:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  25:9   error  Multiple spaces found before 'slidesToShow'    no-multi-spaces
  26:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  26:9   error  Multiple spaces found before 'slidesToScroll'  no-multi-spaces
  27:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  27:7   error  Multiple spaces found before '}'               no-multi-spaces
  28:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  28:5   error  Multiple spaces found before '}'               no-multi-spaces
  29:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  29:5   error  Multiple spaces found before '{'               no-multi-spaces
  30:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  30:7   error  Multiple spaces found before 'breakpoint'      no-multi-spaces
  31:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  31:7   error  Multiple spaces found before 'settings'        no-multi-spaces
  32:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  32:9   error  Multiple spaces found before 'slidesToShow'    no-multi-spaces
  33:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  33:9   error  Multiple spaces found before 'slidesToScroll'  no-multi-spaces
  34:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  34:7   error  Multiple spaces found before '}'               no-multi-spaces
  35:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  35:5   error  Multiple spaces found before '}'               no-multi-spaces
  36:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  37:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  38:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  39:1   error  Mixed spaces and tabs                          no-mixed-spaces-and-tabs
  39:3   error  Multiple spaces found before ']'               no-multi-spaces
  42:1   error  Newline required at end of file but not found  eol-last

✖ 67 problems (67 errors, 0 warnings)

Warning: Task "eslint:target" failed. Use --force to continue.

**Comment by nzakas** [comment]
Please file an issue with pushy
